### ISSUE TITLE
istio gateway never comes up with new istio-dex config

### ISSUE THREAD

**Issue by ryandawsonuk** [kubeflow/kfctl#170]
I updated my istio-dex config to pick up changes, including https://github.com/kubeflow/manifests/commit/f9252596661bbfa7e4bf54c446797c6179b52abf#diff-22a62b4046d14c9056c87c46260349d7

I find that the istio gateway now never becomes available as it fails with `Warning  FailedMount  41s (x23 over 31m)    kubelet, pool-c25fcyqh4-hbx4  MountVolume.SetUp failed for volume "istio-token" : failed to fetch token: the server could not find the requested resource`

**Comment by ryandawsonuk** [comment]
I tried putting [back the old istio config](https://github.com/ryandawsonuk/seldon-gitops/commit/afdb62cc432cae245828827886b1a9c5519b5ae4) and just that change fixed the issue above @krishnadurai

**Comment by yanniszark** [comment]
This seems like an Istio upgrade issue.
/cc @richardsliu do you know what the status of kfctl for upgrades is?

**Comment by ryandawsonuk** [comment]
I found it happened on a newly-created cluster too

**Comment by yanniszark** [comment]
@ryandawsonuk can you link to the guide / kfdef you followed so we can try to reproduce?

**Comment by ryandawsonuk** [comment]
I have been using a custom config but I think it was also happening with the master version of kfctl and https://github.com/kubeflow/manifests/blob/master/kfdef/kfctl_istio_dex.yaml. I'd need to double-check. It was a DigitalOcean cluster if that makes any difference.

**Comment by ryandawsonuk** [comment]
It does work on GKE

**Comment by ryandawsonuk** [comment]
Just tried again on DigitalOcean with `export CONFIG=https://raw.githubusercontent.com/kubeflow/manifests/master/kfdef/kfctl_istio_dex.yaml`. Again it installs but I get the token issue above.

So it works on GKE and not on DigitalOcean. Would be good if somebody could check another cloud.

**Comment by ryandawsonuk** [comment]
It seems problem is https://github.com/istio/istio/issues/17378

**Comment by yanniszark** [comment]
@ryandawsonuk it seems like a DigitalOcean issue.
Do they not support trustworthy JWTs? https://istio.io/blog/2019/trustworthy-jwt-sds/

**Comment by ryandawsonuk** [comment]
Yeah I've raised a ticket with DigitalOcean

**Comment by yanniszark** [comment]
That's great, in that case I believe we can close this issue!
/close

**Comment by swiftdiaries** [comment]
Reopening because I'm facing similar issues on kubeadm created clusters.

**Comment by swiftdiaries** [comment]
One fix is to add extra flags for API server detailed in the Istio KinD [kubeadm configs](https://github.com/istio/istio/blob/master/prow/config/)
Istio issue link: https://github.com/istio/istio/issues/18221

TL;DR: Next release of Istio will enable SDS without those flags
